File: src/main/java/org/jetbrains/space/jenkins/steps/PostBuildStatusAction.java
package org.jetbrains.space.jenkins.steps;

import hudson.model.Action;

import java.io.Serializable;
import java.util.Objects;

public class PostBuildStatusAction implements Action, Serializable {
    private final String spaceConnectionId;
    private final String projectKey;
    private final String repositoryName;
    private final String branch;
    private final String revision;

    public PostBuildStatusAction(String spaceConnectionId, String projectKey, String repositoryName, String branch, String revision) {
        this.spaceConnectionId = spaceConnectionId;
        this.projectKey = projectKey;
        this.repositoryName = repositoryName;
        this.branch = branch;
        this.revision = revision;
    }

    public String getSpaceConnectionId() {
        return spaceConnectionId;
    }

    public String getProjectKey() {
        return projectKey;
    }

    public String getRepositoryName() {
        return repositoryName;
    }

    public String getBranch() {
        return branch;
    }

    public String getRevision() {
        return revision;
    }

    @Override
    public String getIconFileName() {
        return null;
    }

    @Override
    public String getDisplayName() {
        return "Post build status to JetBrains Space";
    }

    @Override
    public String getUrlName() {
        return null;
    }
}

File: src/main/java/org/jetbrains/space/jenkins/steps/PostBuildStatusAction.kt
package org.jetbrains.space.jenkins.steps

import hudson.model.Action
import java.io.Serializable

/**
 * An implementation of [hudson.model.Action] that is stored in build run metadata
 * to notify the build completion listener [RunListenerImpl] that build status reporting has already happened
 * and it shouldn't post the final status on build completion, overwriting the one provided by this step
 */
class PostBuildStatusAction(
    val repositoryName: String,
    val branch: String,
    val revision: String
) : Action, Serializable {

    override fun getIconFileName() = null
    override fun getDisplayName() = "Post build status to JetBrains SpaceCode"
    override fun getUrlName() = null
}
